shared/ui/src/commonMain/kotlin/com/androidmakers/ui/common/navigation/UserViewModel.kt

```diff
@@ -0,0 +1,9 @@
+package com.androidmakers.ui.common.navigation
+
+import fr.androidmakers.domain.repo.UserRepository
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+
+class UserData: KoinComponent {
+  val userRepository: UserRepository by inject()
+}
```

Comment:
I'm not mega proud about this... The underlying problem is I didn't find a way to pass a viewModel all the way down to `onActivityResult` (which is not @Composable). Ideas welcome.